tokio/src/runtime/queue.rs

```diff
@@ -558,11 +606,30 @@ impl<T: 'static> Drop for Inject<T> {
 }
 
 fn get_next(header: NonNull<task::Header>) -> Option<NonNull<task::Header>> {
-    unsafe { *header.as_ref().queue_next.get() }
+    unsafe { header.as_ref().queue_next.with(|ptr| *ptr) }
 }
 
 fn set_next(header: NonNull<task::Header>, val: Option<NonNull<task::Header>>) {
     unsafe {
-        *header.as_ref().queue_next.get() = val;
+        header.as_ref().queue_next.with_mut(|ptr| *ptr = val);
     }
 }
+
+/// Split the head value into the real head and the index a stealer is workin
+/// on.
+fn split(n: u16) -> (u8, u8) {
+    let real = n & u8::max_value() as u16;
+    let steal = n >> 8;
+
+    (steal as u8, real as u8)
+}
+
+/// Join the two head values
+fn join(steal: u8, real: u8) -> u16 {
```

Comment:
nit/TIOLI: elsewhere we refer to the two head values together as "packed". might be a little clearer if these were named `unpack` and `pack`? not a blocker.